src/main/java/ti4/commands2/player/SCPick.java

```diff
@@ -234,21 +238,31 @@ public static void secondHalfOfSCPick(GenericInteractionCreateEvent event, Playe
             }
 
         } else {
-            if (allPicked) {
-                ListTurnOrderService.turnOrder(event, game);
-            }
             if (!allPicked) {
                 game.updateActivePlayer(privatePlayer);
                 MessageHelper.sendMessageToChannelWithButtons(event.getMessageChannel(), msgExtra + "\nUse buttons to pick your strategy card.", Helper.getRemainingSCButtons(game, privatePlayer));
                 game.setPhaseOfGame("strategy");
             } else {
                 MessageHelper.sendMessageToChannel(game.getMainGameChannel(), msgExtra);
+                if (game.isShowBanners()) {
+                    BannerGenerator.drawPhaseBanner("action", game.getRound(), game.getActionsChannel());
+                }
+                game.setPhaseOfGame("action");
+                ListTurnOrderService.turnOrder(event, game);
                 privatePlayer.setInRoundTurnCount(privatePlayer.getInRoundTurnCount() + 1);
                 if (game.isShowBanners()) {
                     BannerGenerator.drawFactionBanner(privatePlayer);
                 }
-                MessageHelper.sendMessageToChannelWithButtons(game.getMainGameChannel(), "\n Use Buttons to do turn.",
-                    StartTurnService.getStartOfTurnButtons(privatePlayer, game, false, event));
+                String text = player.getRepresentationUnfogged() + ", it is now your turn (your " 
+                    + StringHelper.ordinal(player.getInRoundTurnCount()) + " turn of round " + game.getRound() + ").";
+                Player nextPlayer = EndTurnService.findNextUnpassedPlayer(game, player);
+                if (nextPlayer == null || game.isFowMode()) {
```

Comment:
It's a code-smell to have empty if statements like this. I'd change it to:

```            if (nextPlayer != null && !game.isFowMode()) {
                if (nextPlayer == player) {
                    text += "\n-# All other players are passed; you will take consecutive turns until you pass, ending the action phase.";
                } else {
                    text += "\n-# " + nextPlayer.getRepresentationNoPing() + " will start their turn once you've ended yours.";
                }
            }```

Reply:
Fixed.